### PR TITLE
ARC-2723 add share modal 

### PR DESCRIPTION
**Pull request by rachellerathbone**
**What's in this PR?**
Added onClick event to open a modal for the share page button.

**Why**
So Jira admins can share a link to the global page with their Jenkins admin.

In the following recording:

- I click the share page button which opens the modal
- I click copy to clipboard and see a confirmation
- I copy the contents to a Confluence page (demo purposes)
- I show that there are 2 people with access to the jira site: admin Rachelle and non-admin Rachelle 🙃 
- Non Jira admin Rachelle pastes the link and is able to view the global page (this will only work in dev atm coz it's deployed there... maybe stg too, I can't remember 🤷 )

https://github.com/atlassian/jenkins-for-jira/assets/37155488/12cad489-c648-45f0-ae7e-24053ce53e90

**Added feature flags**
always the same

**Affected issues**  
ARC-2729

**How has this been tested?**  
Locally and unit tests

**What's Next?**
Merge many many PRs so @joshkay10 doesn't have to wrap up all my work next week? 🤔

### CODE REVIEW

File: app/jenkins-for-jira-ui/src/components/JenkinsServerList/ConnectedServer/JenkinsModal.tsx [hunk 1]
```diff
@@ -1,26 +1,32 @@
...
 import Modal, {
-	Appearance,
 	KeyboardOrMouseEvent,
 	ModalBody,
 	ModalFooter,
 	ModalHeader,
 	ModalTitle,
-	ModalTransition
+	ModalTransition,
+	Appearance as ModalAppearance
```
Comment (rachellerathbone) [inline app/jenkins-for-jira-ui/src/components/JenkinsServerList/ConnectedServer/JenkinsModal.tsx:10]: Updated this so we were using the correct type for the modal appearance. Previously we were pulling in the type from the Button component which does not give the correct types.

File: app/jenkins-for-jira-ui/src/components/JenkinsServerList/ConnectedServer/JenkinsModal.tsx [hunk 2]
```diff
@@ -44,35 +51,61 @@ const JenkinsModal: React.FC<ModalProps> = ({
...
 	return (
 		<ModalTransition>
 			{show && (
 				<Modal onClose={onClose} testId={dataTestId}>
 					<ModalHeader>
-						<ModalTitle appearance={modalAppearance}>{title}</ModalTitle>
+						{
+							modalAppearance
+								? (
+									<ModalTitle appearance={modalAppearance}>{title}</ModalTitle>
```
Comment (rachellerathbone) [inline app/jenkins-for-jira-ui/src/components/JenkinsServerList/ConnectedServer/JenkinsModal.tsx:65]: The only 2 modal appears are warning/danger so we only want to apply for some use cases.

File: app/jenkins-for-jira-ui/src/components/ServerManagement/ServerManagement.styles.tsx [hunk 3]
```diff
@@ -1,11 +1,43 @@
 import { css } from '@emotion/css';
 import { token } from '@atlaskit/tokens';
 
-export const mainPageContainer = css`
+export const serverManagementContainer = css`
```
Comment (rachellerathbone) [inline app/jenkins-for-jira-ui/src/components/ServerManagement/ServerManagement.styles.tsx:4]: Missed this guy in my previous refactor to rename 'MainPage'

File: app/jenkins-for-jira-ui/src/components/ServerManagement/ServerManagement.tsx [hunk 4]
```diff
@@ -1,34 +1,120 @@
...
+			<Button appearance="primary">Connect a new Jenkins server</Button>
+			<Button onClick={() => handleShowSharePageModal()}>Share page</Button>
 		</ButtonGroup>
 	);
 
-	// TODO handle empty state - ARC-2730 connection wizard
+	const sharePageMessage =
+		`Hi there,
+
+Jenkins for Jira is now installed and connected on ${getSiteNameFromUrl(globalPageUrl)}.
```
Comment (rachellerathbone) [inline app/jenkins-for-jira-ui/src/components/ServerManagement/ServerManagement.tsx:75]: Alignment for this looks whack but, unfortunately, I don't think there's a way around it. Indenting means indenting the content in the text area and everything needs to be left-aligned.

File: app/jenkins-for-jira-ui/src/components/JenkinsServerList/ConnectedServer/JenkinsModal.tsx [hunk 5]
```diff
@@ -1,26 +1,32 @@
...
 	ModalHeader,
 	ModalTitle,
-	ModalTransition
+	ModalTransition,
+	Appearance as ModalAppearance
 } from '@atlaskit/modal-dialog';
-import { Appearance as ButtonAppearance } from '@atlaskit/button';
+import EditorSuccessIcon from '@atlaskit/icon/glyph/editor/success';
+import { cx } from '@emotion/css';
 import { JenkinsServer } from '../../../../../src/common/types';
```
Comment (joshkay10) [inline app/jenkins-for-jira-ui/src/components/JenkinsServerList/ConnectedServer/JenkinsModal.tsx:14]: lol lotta ././././././ not now but sometime, might be time we use aliases like GH4J

**Review by joshkay10 — APPROVED** [review]

**Pull request merged**